src/k2/module/k23si/Module.cpp

```diff
@@ -756,7 +751,7 @@ bool K23SIPartitionModule::_makeFieldsForDiffVersion(dto::Schema& schema, dto::S
     return true;
 }
 
-bool K23SIPartitionModule::_parsePartialRecord(dto::K23SIPartialUpdateRequest& request, std::deque<dto::DataRecord>& versions) {
+bool K23SIPartitionModule::_parsePartialRecord(dto::K23SIWriteRequest& request, std::deque<dto::DataRecord>& versions) {
     auto schemaIt = _schemas.find(request.key.schemaName);
     if (schemaIt == _schemas.end()) return false;
     auto schemaVer = schemaIt->second.find(request.value.schemaVersion);
```

Comment:
I think this schema validation should happen via helper in handleWrite() and we should return a status indicating that schema wasn't found

Reply:
Yes, handleWrite was already checking for the schema to exist, so I removed the checks here.